### PR TITLE
Add support for project datatype in custom fields

### PR DESCRIPTION
**Pull request by oveaurs**
https://github.com/ankitpokhrel/jira-cli/discussions/481

### CODE REVIEW

File: pkg/jira/edit.go [hunk 1]
```diff
@@ -323,6 +323,8 @@ func constructCustomFieldsForEdit(fields map[string]string, data *editRequest) {
 			switch configured.Schema.DataType {
 			case customFieldFormatOption:
 				data.Update.M.customFields[configured.Key] = []customFieldTypeOptionSet{{Set: customFieldTypeOption{Value: val}}}
+			case customFieldFormatProject:
+				data.Update.M.customFields[configured.Key] = []customFieldTypeFormatSet{{Set: customFieldTypeFormat{Value: val}}}
```
Comment (yurrriq) [inline pkg/jira/edit.go:327]: maybe this?

```suggestion
				data.Update.M.customFields[configured.Key] = []customFieldTypeProjectSet{{Set: customFieldTypeProject{Value: val}}}
```

Comment (oveaurs) [reply pkg/jira/edit.go]: yes, a bit quick on the pushing there :)

File: pkg/jira/customfield.go [hunk 2]
```diff
@@ -1,9 +1,10 @@
 package jira
 
 const (
-	customFieldFormatOption = "option"
-	customFieldFormatArray  = "array"
-	customFieldFormatNumber = "number"
+	customFieldFormatOption  = "option"
+	customFieldFormatArray   = "array"
+	customFieldFormatNumber  = "number"
+	customFieldFormatProject = "project"
```
Comment (ankitpokhrel) [inline pkg/jira/customfield.go:7]: Hi @oveaurs, I don't see this field format in the [Jira doc](https://developer.atlassian.com/cloud/jira/service-desk/rest/intro/#fieldformats). I am wondering what's the datatype for this field returned by the [`/field`](https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-fields/#api-rest-api-2-field-get) API?

Comment (oveaurs) [reply pkg/jira/customfield.go:7]:
```{
    "id": "customfield_13467",
    "name": "Cross over project",
    "custom": true,
    "orderable": true,
    "navigable": true,
    "searchable": true,
    "clauseNames": [
      "cf[13467]",
      "Cross over project"
    ],
    "schema": {
      "type": "project",
      "custom": "com.atlassian.jira.plugin.system.customfieldtypes:project",
      "customId": 13467
    }
  },
```
I found a reference to that datatype here: https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-fields/

Comment (ankitpokhrel) [reply pkg/jira/customfield.go:7]: I see, thank you for the reference. The changes looks OK to me.

**Review by ankitpokhrel — APPROVED** [review]

**Comment by ankitpokhrel** [comment]
Thank you for your contribution 🎉

**Pull request merged**